AndrewTehan/hometask-3/rainfall_statistics.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# this class is used for calculate mean and variance values for town rainfall
+class RainfallStatistics
+  attr_reader :town, :data
+
+  def initialize(town, data)
+    @town = town
+    @data = data
+  end
+
+  def mean
+    return -1 if missing_data?
+
+    average(extract_statistic_values)
+  end
+
+  def variance
+    return -1 if missing_data?
+
+    calculate_variance(calculate_mean_differences)
+  end
+
+  private
+
+  def extract_statistic_values
+    @extract_statistic_values ||= find_town_statistics.scan(/[0-9]*\.[0-9]*/).map(&:to_f)
```

Comment:
You're lucky that you have an easy regular expression like `/[0-9]*\.[0-9]*/` and I could guess what it does. But sometimes it's very hard to get a purpose of regexp.

I'd recommend avoiding using NONAME stuff, especially when it could not be clear what it's for. Let's move it to a constant 💃🏻